### PR TITLE
Using the parameter argc instead of calculating the size of the vector arguments

### PR DESCRIPTION
**Pull request by 31Nathan**
In the ArgumentsParser Class, add a parameter to the constructor (argc) to directly have the arguments number to avoid calculating it

### CODE REVIEW

**Review by osvegn — APPROVED** [review]
LGTM

File: src/Parser/ArgumentsParser/ArgumentsParser.hpp [hunk 1]
```diff
@@ -15,8 +15,10 @@
 /// \brief ArgumentsParser class define functions for parsing received arguments
 class ArgumentsParser : public AParser {
     public:
-        /// \brief Construct the ArgumentsParser with the received arguments without the binary name (argv + 1)
-        ArgumentsParser(char ** = nullptr);
+        /// \brief Construct the ArgumentsParser with the received arguments
+        /// \param ac number of arguments without the binary name (ac - 1)
+        /// \param av arguments tab without the binary name (av + 1)
+        ArgumentsParser(int = 0, char ** = nullptr);
```
Comment (Cavonstavant) [inline src/Parser/ArgumentsParser/ArgumentsParser.hpp:21]: Add a default constructor and add explicit before parameterized constructor

**Review by Cavonstavant — CHANGES_REQUESTED** [review]
minor changes needed

File: src/Parser/ArgumentsParser/ArgumentsParser.hpp [hunk 2]
```diff
@@ -15,8 +15,12 @@
 /// \brief ArgumentsParser class define functions for parsing received arguments
 class ArgumentsParser : public AParser {
     public:
-        /// \brief Construct the ArgumentsParser with the received arguments without the binary name (argv + 1)
-        ArgumentsParser(char ** = nullptr);
+        /// \brief Default Constructor of the ArgumentsParser DO NOT USE
+        ArgumentsParser();
```
Comment (Cavonstavant) [inline src/Parser/ArgumentsParser/ArgumentsParser.hpp:19]: missing = default();

**Review by Cavonstavant — CHANGES_REQUESTED** [review]

**Review by Cavonstavant — APPROVED** [review]
LGTM!

**Pull request merged**